pkg/shp/cmd/build/delete.go

```diff
@@ -0,0 +1,60 @@
+package build
+
+import (
+	"errors"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/cli/pkg/shp/cmd/runner"
+	"github.com/shipwright-io/cli/pkg/shp/params"
+	"github.com/spf13/cobra"
+)
+
+// DeleteSubCommand contains data provided by user to the delete subcommand
+type DeleteSubCommand struct {
+	name string
+
+	cmd *cobra.Command
+}
+
+func deleteSubCmd() runner.SubCommand {
+	c := &cobra.Command{
+		Use:   "delete [flags] name",
+		Short: "Delete Build",
+	}
+
+	return &DeleteSubCommand{
+		cmd: c,
+	}
+}
+
+// Cmd returns cobra command object of the delete subcommand
+func (sc *DeleteSubCommand) Cmd() *cobra.Command {
+	return sc.cmd
+}
+
+// Complete fills DeleteSubCommand structure with data obtained from cobra command
+func (sc *DeleteSubCommand) Complete(params params.Params, args []string) error {
+	if len(args) < 1 {
+		return errors.New("Missing 'name' argument")
+	}
+
+	sc.name = args[0]
+
+	return nil
+}
+
+// Validate is used for validation of user input data
+func (sc *DeleteSubCommand) Validate() error {
+	return nil
+}
+
+// Run contains main logic of delete subcommand
+func (sc *DeleteSubCommand) Run(params params.Params) error {
+	var b buildv1alpha1.Build
+
+	if err := buildResource.Get(sc.name, &b); err != nil {
+		return err
```

Comment:
Wouldn't it be better to notify user, that thing they are trying to delete doesn’t exist, rather than silently do nothing? If user accidentally makes a typo in build name, client will tell them, that something is wrong. Opposed to making it look like everything was ok, and user might think that build has been deleted.
@adambkaplan @otaviof what do you think?

Reply:
Yeah, I might be thinking of how we did it internal to OpenShift. If OpenShift created something, then went to delete it and it was already gone, then we didn't care. But if the user does it, and it fails, we probably should let them know.  I consider this one resolved.